DOCUMENTATION.md

```diff
@@ -107,19 +119,138 @@ In case it's a **dictionary** with fields, please make sure the **encoder** func
 - `confirmation`: **boolean**  this boolean indicates if you wish to receive a confirmation after sending the downlink message. Default value to False.
 - `schedule`: **string**  this string provides the type of schedule on which the message should be sent. It can take values such as `first` or `last`. Default value to `replace`.
 
-### Errors
-Errors can happen on connection for different reasons:
+### UplinkMessage
+This type of object is constructed dynamically from the message received by the client, so this means some attributes can change from one message to another. However here are some constant attributes usually found in UplinkMessage objects:
+* `app_id`: the application ID to which the device is registered
+* `dev_id`: the ID of the device
+* `port`: the port number on which the message was sent
+* `payload_raw`: a buffer which contains the payload in hexadecimal
+* `metadata`: this field is another object which contains all the metadata of the message. Such as: the date, the frequency, the data rate and the list of gateways.
+
+
+## ApplicationClient
+The class constructor can be called following this scheme:
+```python
+ApplicationClient(app_id, access_key_or_token, [net_address], [certificate], [discovery_address])
+```
+- `app_id`: **string**  this the name given to the application when it was created.
+![Screenshot of the console with app section](./images/app-console.png?raw=true)
+- `access_key_or_token`: **string**  this can be found at the bottom of the application page under **ACCESS KEYS**. You will need a key allowing you to change the settings if you wish to update your application.
+- `net_address`: **string**  this  is the address of the handler to which the application was registered. It needs to be provided as a `net_address=value` argument when calling the constructor.
+- `certificate`: **string**  this is the certificate used to connect in a secure way to the handler. It needs to be provided as a `certificate=value` argument when calling the constructor.
```

Comment:
Can you be more specific whether the `certificate` is the actual content of the certificate, or a link to it (URI, filepath)?